File: Go/abb_menu/abb_program.go
package main

import "fmt"

func preorderWithBracket(a ABB) {
	fmt.Printf("(")
	if !isEmptyTree(a){
		if !isEmptyTree(leftSon(a)) || !isEmptyTree(rightSon(a)){
			fmt.Printf("%s%d%s"," ",key(a)," ")
			preorderWithBracket(leftSon(a))
			fmt.Printf(" ")
			preorderWithBracket(rightSon(a))
		}else{
			fmt.Printf("%s%d%s"," ",key(a)," ")
		} 
	}
	
	fmt.Printf(")")
}

func main() {
	var abb ABB
<<<<<<< Updated upstream:Go/abb_menu/abb_program.go
	abb = &tNodeA{4,nil,nil}
	emptyTree(&abb)

	insertKey(&abb,4)
	insertKey(&abb,4)
	insertKey(&abb,2)
	insertKey(&abb,6)
	insertKey(&abb,1)
	insertKey(&abb,3)
	insertKey(&abb,5)
	insertKey(&abb,7)
	preorderWithBracket(abb)
	fmt.Printf("\n")
=======
	emptyTree(abb)
	if abb == nil{
		fmt.Printf("NULO")
	}
	insertKey(abb,5)
	if abb == nil{
		fmt.Printf("NULO")
	}else{
		preorderWithBracket(abb)
	}
>>>>>>> Stashed changes:Go/src/abb_menu/abb_program.go
}

File: DLP/P1/Lua/abb_program.lua

-- Autor1: Yeray Méndez Romero
-- Login1: yeray.mendez
-- Email1: yeray.mendez@udc.es
-- Autor2: Daniel Rivera López
-- Login2: d.rivera1
-- Email2: d.rivera1@udc.es


local abbTad = require("TAD")

function preorderWithBracket(abb)
    io.write('(')
	if not isEmptyTree(abb) then
		if not isEmptyTree(leftSon(abb)) or  not isEmptyTree(rightSon(abb)) then
			io.write(' ',key(abb),' ')
			preorderWithBracket(leftSon(abb))
			io.write(' ')
			preorderWithBracket(rightSon(abb))
		else
			io.write(' ',key(abb),' ')
        end
    end
	io.write(')')
end

function test()
	local tree
	tree = emptyTree(nil)
	tree = insertKey(tree,4)
	tree = insertKey(tree,4)
	tree = insertKey(tree,2)
	tree = insertKey(tree,6)
	tree = insertKey(tree,1)
	tree = insertKey(tree,3)
	tree = insertKey(tree,5)
	tree = insertKey(tree,7)

	preorderWithBracket(tree)
	print()
		
	print('search 1...',key(searchKey(tree,1)))
	print('search 2...',key(searchKey(tree,2)))
	print('search 3...',key(searchKey(tree,3)))
	print('search 4...',key(searchKey(tree,4)))
	print('search 5...',key(searchKey(tree,5)))
	print('search 6...',key(searchKey(tree,6)))
	print('search 7...',key(searchKey(tree,7)))

	print('remove 5...')
	tree = eraseKey(tree,5)
	preorderWithBracket(tree)
	print()

	print('remove 6...')
	tree = eraseKey(tree,6)
	preorderWithBracket(tree)
	print()

	print('remove 4...')
	tree = eraseKey(tree,4)
	preorderWithBracket(tree)
	print()

	print('remove 2...')
	tree = eraseKey(tree,2)
	preorderWithBracket(tree)
	print()
end